cmd/init.go

```diff
@@ -92,32 +95,59 @@ setup the local dev environment.`,
 			if !repos.Has(repoName) {
 				return errors.Errorf("Repository %s is not in configured list of repositories", repoName)
 			}
+			var templateName string
+			var inputTemplateName string
+			if len(args) >= 2 {
+
+				inputTemplateName = args[1]
+				if inputTemplateName == "none" {
+					noTemplate = true
+				}
+
+			}
+
+			templateName = inputTemplateName // so we can keep track
 
 			Debug.log("Attempting to locate stack ", projectType, " in repo ", repoName)
 			index = indices[repoName]
 			projectFound := false
 			stackFound := false
+
 			if strings.Compare(index.APIVersion, supportedIndexAPIVersion) == 1 {
 				Warning.log("The repository .yaml for " + repoName + " has a more recent APIVersion than the current Appsody CLI supports (" + supportedIndexAPIVersion + "), it is strongly suggested that you update your Appsody CLI to the latest version.")
 			}
 			if len(index.Projects[projectType]) >= 1 { //V1 repos
 				projectFound = true
 				//return errors.Errorf("Could not find a stack with the id \"%s\" in repository \"%s\". Run `appsody list` to see the available stacks or -h for help.", projectType, repoName)
 				Debug.log("Project ", projectType, " found in repo ", repoName)
+
+				// need to check template name vs default?
 				projectName = index.Projects[projectType][0].URLs[0]
 			}
-
 			for _, stack := range index.Stacks {
 				if stack.ID == projectType {
 					stackFound = true
 					Debug.log("Stack ", projectType, " found in repo ", repoName)
-					projectName = stack.Templates[0].URL
+					URL := ""
+					if templateName == "" || templateName == "none" {
+						templateName = stack.DefaultTemplate
```

Comment:
Why are we getting default template for "none"?

Reply:
Because it still needs to extract .appsody-config.yaml, which brings up good point.  Because which template do we want if we remove the --no-template option.

Reply:
We believe we are good as is.  There is a performance improvement to be gained by generating .appsody-config.yaml, but that will be in a future sprint.